controller/volume_controller.go

```diff
@@ -866,11 +866,11 @@ func (vc *VolumeController) cleanupExtraHealthyReplicas(v *longhorn.Volume, e *l
 		return err
 	}
 
-	if cleaned, err = vc.cleanupAutoBalancedReplicas(v, e, rs); err != nil || cleaned {
+	if cleaned, err = vc.cleanupDataLocalityReplicas(v, e, rs); err != nil || cleaned {
 		return err
 	}
 
-	if cleaned, err = vc.cleanupDataLocalityReplicas(v, e, rs); err != nil || cleaned {
+	if cleaned, err = vc.cleanupAutoBalancedReplicas(v, e, rs); err != nil || cleaned {
 		return err
```

Comment:
Why exchanging the order can fix the issue?

Reply:
The issue manifested because the data locality is trying to rebuild a new replica, and it got cleaned up by the auto-balance because of the sorted order. 

Exchanging the order will let the data locality clean up its own redundant replica.

Reply:
Updated the description.

Reply:
Got it, thank you.
Can we add a comment here for avoiding running into the issue when others work on it in the future?

Reply:
Sure, updated.